src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -1260,9 +1263,10 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
             },
         };
 
-        // Load supported emojis into registry.
+        // Load supported coin symbol emojis into registry.
+        let coin_symbol_emojis_ref_mut = &mut registry.coin_symbol_emojis;
```

Comment:
Would need to look more into the disassembled bytecode, but I have a hunch doing this declaration before the `for_each_ref` loop will avoid the gas cost associated with field access that was previously inside each iteration (e.g. assuming compiler isn't optimizing this out)

Reply:
Oh interesting- you're saying you think each loop iteration is charged for a field access if you use `&mut registry.coin_symbol_emojis` as opposed to `let coin_symbol_emojis_ref_mut = &mut registry.coin_symbol_emojis;` before the loop?

Reply:
That's my hunch

Before a few days ago I would've assumed that it is optimized out

But now I suspect the compiler has no optimizations